### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by RICKCOYL**
### In this set of iterations, we

- Designed an entity-relationship diagram

- Implemented User, Post, and Comment models

- Added validations for the user, post and comment model

- Added associations to the models

- Added a comment model

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  validates :name, presence: true, length: { maximum: 50 }
+
+  validates :name, uniqueness: true
+
+  has_many :post
+
+  has_many :comment
```
Comment (peterrobert) [inline app/models/user.rb]: These relationships are wrong. Kindly check below :arrow_down: 

- [ ] A user has many posts not post.

- [ ] A user has many comments not comment

### hint
Always remember when we are talking about has_many the model you are referring to should always be pluralized

posts

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+  validates :title, uniqueness: true
+
+  validates :body, presence: true, length: { maximum: 50 }
+
+  validates :body, uniqueness: true
+
+  belongs_to :user
+
+  has_many :comment
```
Comment (peterrobert) [inline app/models/post.rb]: This relationship has some issues. Check below :arrow_down: 

- [ ] A post has many comments not comment

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: :stop_sign: 
HELLO @RICKCOYL  AND @Daniduran-dev  :wave:  VERY NICE PROGRESS. YOU HAVE DONE A GOOD JOB. HOWEVER, THERE RE A FEW CHANGES TO BE MADE KINDLY IMPROVE ON THE FOLLOWING :arrow_down: 

- [ ] Kindly add a front end to your application

### if you have any question about the review kindly reach me on slack @peterrobert

### HAPPY CODING :computer:

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 
## Great job guys your project is fully functional.

now you can merge to develop your project is approved.

## Happy Coding :tophat:

**Pull request merged**